# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1231781769 18000
#      Mon Jan 12 12:36:09 2009 -0500
# Node ID cb48e594a441bc168abcc06b07b8a15c3b8f5bce
# Parent  a74871446af73598f8f6ef5811ef5643d9a5e4f7
Fchar, Fstrvcat: use queue correctly

diff --git a/src/strfns.cc b/src/strfns.cc
--- a/src/strfns.cc
+++ b/src/strfns.cc
@@ -107,17 +107,18 @@ char ([97, 98, 99], \"\", @{\"98\", \"99
 	}
 
       string_vector result (n_elts);
 
       int k = 0;
 
       for (int i = 0; i < nargin; i++)
 	{
-	  string_vector s = args_as_strings.pop ();
+	  string_vector s = args_as_strings.front ();
+	  args_as_strings.pop ();
 
 	  int n = s.length ();
 
           if (n > 0)
             {
 	      for (int j = 0; j < n; j++)
 	        {
 	          std::string t = s[j];
@@ -226,17 +227,18 @@ strvcat ([97, 98, 99], \"\", @{\"98\", \
 	}
 
       string_vector result (n_elts);
 
       int k = 0;
 
       for (int i = 0; i < nargin; i++)
 	{
-	  string_vector s = args_as_strings.pop ();
+	  string_vector s = args_as_strings.front ();
+	  args_as_strings.pop ();
 
 	  int n = s.length ();
 
           if (n > 0)
             {
 	      for (int j = 0; j < n; j++)
 	        {
 	          std::string t = s[j];
